### PR TITLE
Add support for the quality parameter (and a refactor)

### PR DESCRIPTION
**Pull request by rowanmanning**

### CODE REVIEW

File: lib/image-service-url.js [hunk 1]
```diff
@@ -6,20 +6,103 @@ module.exports = class ImageServiceUrl {
...
+	}
+
+	_setNumericProperty(property, value) {
+		if (value === undefined) {
+			return this[property] = value;
+		}
+		if (typeof value !== 'string' && typeof value !== 'number') {
+			throw new Error(`Invalid ${property} parameter`);
+		}
+		value = Number(value);
```
Comment (onishiweb) [inline lib/image-service-url.js:60]: What are our conventions re block statements and new lines. I had a look in the ESLint rules but nothing stood out. Just curious, as I almost missed this `value = Number(value);` line between the multiple blocks and wondered whether a new line after blocks would be good?

**Comment by AlbertoElias** [comment]
This looks great! 👍

File: lib/image-service-url.js (hunk 1 above)
Comment (rowanmanning) [inline lib/image-service-url.js:60]: I'm not sure if we have conventions for that, @AlbertoElias?

Comment (AlbertoElias) [inline lib/image-service-url.js:60]: No conventions, I usually add new lines, but definitely not consistently

Comment (onishiweb) [inline lib/image-service-url.js:60]: Would be great to add something on this if people are ok with it? A little whitespace around blocks is really useful for readability. :)

Comment (rowanmanning) [inline lib/image-service-url.js:60]: OK gonna merge and will revise when we come up with a standard :) maybe a good discussion point in the next Origami meeting? We don't even use half of ESLint's capabilities

Comment (onishiweb) [inline lib/image-service-url.js:60]: Yep works for me :)

**Pull request merged**